tests/python/openassetio/managerAPI/test_hostsession.py

```diff
@@ -40,14 +41,39 @@ def mock_logger():
 
 
 @pytest.fixture
-def host_session(mock_host, mock_logger):
-    return HostSession(mock_host, mock_logger)
+def host_session(a_host, mock_logger):
+    return HostSession(a_host, mock_logger)
 
 
-class TestHostSession():
+class Test_HostSession_init:
+    def test_when_invalid_host_then_raises_TypeError(self, mock_logger):
+        with pytest.raises(TypeError) as err:
+            HostSession(object(), mock_logger)
+
+        assert str(err.value).startswith("__init__(): incompatible constructor arguments")
+
+
+class Test_HostSession_host:
+    def test_returns_copy_of_host(self, host_session, a_host, mock_host_interface):
```

Comment:
[fix] This is testing that we return by value in cpp. Is the title of this test the most descriptive? There isn't any particularly reason that it _should_ be a copy, we just choose to do so as it's a simple stateless container that can trivially be handled by value. From a TDD perspective it perhaps makes it sound like there is some ulterior motive behind it being a copy, and that that is required for proper functioning of the system.

Reply:
Could it be seen as documenting the behaviour?  I.e. users should expect the host retrieved from the session to be fresh, i.e. any changes they make to the host instance won't be persisted.

E.g. if the user does `host.my_variable = 123` then retrieves `.host()` again (or after passing `host` to the `HostSession` constructor), `my_variable` will be lost. 

IIRC hanging new variables off an instance is disallowed by default with pure-pybind instances, so removing the Python `Host` will mitigate that.

How about `test_returns_new_host_wrapper`?

Reply:
Do we even need this test? It seems like its not testing system behaviour but instead some specifics of how we currently have the language bindings. All we really care about is the host supplied "the same" as the one that went into it.

Reply:
Updated to use `SharedPtr` in 3ca11608 - certainly makes this test a lot simpler!